src/transactions/test/TxEnvelopeTests.cpp

```diff
@@ -2513,13 +2513,6 @@ TEST_CASE("soroban transaction validation", "[tx][envelope][soroban]")
         }
     };
 
-    SECTION("no soroban extension")
```

Comment:
Why is this removed?

Reply:
We shouldn't allow malformed transactions to get this far (to check valid), so the extension check moved up in the flow, and now we reject malformed transactions at the tx queue stage, hence I moved this test case to `"Soroban TransactionQueue limits"`

Reply:
> We shouldn't allow malformed transactions to get this far (to check valid)

I think that's a wrong assumption. When we're getting a tx set, we need to call `checkValid` on every transaction, so a malformed tx set would now crash Core apparently (we probably don't have a test for this particular error condition). In general, I think it's not correct to make any assumptions on the flows that are external to tx frame - if it has a public function, it should be well-behaved in all the appropriate contexts.